src/main/java/dev/dfonline/codeclient/dev/highlighter/ExpressionHighlighter.java

```diff
@@ -0,0 +1,350 @@
+package dev.dfonline.codeclient.dev.highlighter;
+
+import dev.dfonline.codeclient.CodeClient;
+import dev.dfonline.codeclient.Feature;
+import dev.dfonline.codeclient.config.Config;
+import dev.dfonline.codeclient.hypercube.HypercubeCommands;
+import dev.dfonline.codeclient.hypercube.HypercubeMinimessage;
+import dev.dfonline.codeclient.hypercube.item.VarItem;
+import dev.dfonline.codeclient.hypercube.item.VarItems;
+import dev.dfonline.codeclient.location.Dev;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.format.NamedTextColor;
+import net.kyori.adventure.text.minimessage.MiniMessage;
+import net.kyori.adventure.text.minimessage.tree.Node;
+import net.kyori.adventure.text.serializer.plain.PlainTextComponentSerializer;
+import net.minecraft.client.font.TextRenderer;
+import net.minecraft.client.gui.DrawContext;
+import net.minecraft.predicate.NumberRange;
+import net.minecraft.text.CharacterVisitor;
+import net.minecraft.text.OrderedText;
+import net.minecraft.text.Style;
+import net.minecraft.text.TextColor;
+import net.minecraft.util.Formatting;
+import org.apache.commons.lang3.IntegerRange;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static net.kyori.adventure.platform.fabric.FabricAudiences.nonWrappingSerializer;
+
+public class ExpressionHighlighter extends Feature {
+    private final Set<String> CODES = Set.of(
+            "default",
+            "selected",
+            "uuid",
+            "var",
+            "math",
+            "damager",
+            "killer",
+            "shooter",
+            "victim",
+            "projectile",
+            "random",
+            "round",
+            "index",
+            "entry"
+    );
+
+    // palette copied from recode, maybe there's a better combination?
+    private final List<TextColor> COLORS = List.of(
+            TextColor.fromRgb(0xffd42a),
+            TextColor.fromRgb(0x33ff00),
+            TextColor.fromRgb(0x00ffe0),
+            TextColor.fromRgb(0x5e77f7),
+            TextColor.fromRgb(0xca64fa),
+            TextColor.fromRgb(0xff4242)
+    );
+
+    private final TextColor ERROR_COLOR = TextColor.fromFormatting(Formatting.RED);
+
+    @Override
+    public boolean enabled() {
+        return CodeClient.location instanceof Dev && Config.getConfig().HighlighterEnabled;
```

Comment:
I believe this function is for if it's enabled by the user; not if the state means it's active: https://github.com/DFOnline/CodeClient/blob/d7b56fdd1d56a3a97f140a04516b3b545dca210a/src/main/java/dev/dfonline/codeclient/Feature.java#L21-L27

Reply:
i believe i have fixed this, for some reason i thought this was how the other features used this method